sklearn/preprocessing/_encoders.py

```diff
@@ -383,6 +381,11 @@ def _handle_deprecations(self, X):
                     "The 'categorical_features' keyword is deprecated in "
                     "version 0.20 and will be removed in 0.22. You can "
                     "use the ColumnTransformer instead.", DeprecationWarning)
+                n_features = X.shape[1]
+                sel = np.zeros(n_features, dtype=bool)
+                sel[np.asarray(self.categorical_features)] = True
+                if sum(sel) == 0:
```

Comment:
I have not looked at the code in detail, but isn't this equivalent to,
```py
not self.categorical_features.any()
```
or something similar ?

Reply:
`categorical_features` can be a array of indices or a mask. This works in either case. I followed the pattern in `sklearn.preprocessing.base._transform_selected`

Reply:
Can you add a comment above this code block to explain the specific case that this code is checking for?